src/Microsoft.AspNet.Mvc.Core/ActionDescriptor.cs

```diff
@@ -9,10 +9,25 @@ namespace Microsoft.AspNet.Mvc
 {
     public class ActionDescriptor
     {
+        private readonly Guid _guid;
+
         public ActionDescriptor()
         {
             Properties = new Dictionary<object, object>();
             RouteValueDefaults = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
+            _guid = Guid.NewGuid();
+        }
+
+        /// <summary>
+        /// Gets an id which uniquely identifies the action.
+        /// It is of the format "<see cref="DisplayName"/>_<see cref="Guid"/>".
```

Comment:
The Id should just be the guid. We're also going to be logging the action display name elsewhere